client/js/components/procedure/admin/InstitutionTagManagement/TagList.vue

```diff
@@ -0,0 +1,241 @@
+            <tag-list-item
+              :item="nodeElement"
+              @item:deleted="handleItemDeleted" />
+          </template>
+          <template v-slot:leaf="{ nodeElement }">
+            <tag-list-item
+              :item="nodeElement"
+              @item:deleted="handleItemDeleted" />
+          </template>
+        </dp-tree-list>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import {
+  DpButton,
+  DpInlineNotification,
+  DpLoading,
+  DpTreeList
+} from '@demos-europe/demosplan-ui'
+import { mapActions, mapState } from 'vuex'
+import NewCategoryForm from './NewCategoryForm'
+import NewTagForm from './NewTagForm'
+import TagListItem from './TagListItem'
+
+export default {
+  name: 'TagList',
+
+  components: {
+    DpButton,
+    DpInlineNotification,
+    DpLoading,
+    DpTreeList,
+    NewCategoryForm,
+    NewTagForm,
+    TagListItem
+  },
+
+  data () {
+    return {
+      addNewCategory: false,
+      addNewTag: false,
+      edit: false,
+      editingCategoryId: null,
+      editingTagId: null,
+      isLoading: false,
+      tagCategoriesWithTags: []
+    }
+  },
+
+  computed: {
+    ...mapState('InstitutionTagCategory', {
+      institutionTagCategories: 'items'
+    }),
+
+    ...mapState('InstitutionTag', {
+      institutionTags: 'items'
+    })
+  },
+
+  methods: {
+    ...mapActions('InstitutionTagCategory', {
+      listInstitutionTagCategories: 'list'
+    }),
+
+    closeNewCategoryForm () {
+      this.addNewCategory = false
+    },
+
+    closeNewTagForm () {
+      this.addNewTag = false
+    },
+
+    getInstitutionTagCategories () {
+      this.isLoading = true
+
+      this.listInstitutionTagCategories({
+        fields: {
+          InstitutionTagCategory: [
+            'name',
+            'tags'
+          ].join(),
+          InstitutionTag: [
+            'isUsed',
+            'name'
+          ].join()
+        },
+        include: [
+          'tags'
+        ].join()
+      })
+        .then(() => {
+          this.isLoading = false
+          this.tagCategoriesWithTags = this.transformTagsAndCategories()
+        })
+        .catch(err => {
+          console.error(err)
+          this.isLoading = false
+        })
+    },
+
+    handleAddNewCategoryForm () {
+      this.addNewCategory = true
+      this.editingCategoryId = null
+    },
+
+    handleAddNewTagForm () {
+      this.addNewTag = true
+      this.editingTagId = null
+    },
+
+    /**
+     * Remove item from the list
+     * @param item { object } With properties id, name, type; if it's a category, also children;
+     * if it's a tag, also categoryId and isUsed
+     */
+    handleItemDeleted (item) {
+      const isTag = !!item.categoryId
+      const isCategory = !item.categoryId
+
+      if (isTag) {
+        const category = this.tagCategoriesWithTags.find(category => category.id === item.categoryId)
+
+        category.children = category.children.filter(tag => tag.id !== item.id)
```

Comment:
Is there a scenario where `.find()` can be undefined? If so then this code would break in that case.

Reply:
In the interface, it is not possible to create a tag without assigning it to a category, so "under normal circumstances", it should not come to this scenario. Do you think we should still handle it?

Reply:
If it is logically impossible I think it is good. However if we want to run the extra mile we could wrap it in a` try{}-catch{}`-block but this would be more of a cosmetic upgrade in the end :)